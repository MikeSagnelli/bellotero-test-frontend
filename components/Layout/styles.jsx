import styled from 'styled-components';
import { whiteColor, primaryColor, containerSize } from '../../static/styleConstants';

export const MainContainer = styled.div`
  width: 100%;
  height: 100vh;
`;

export const Header = styled.div`
  display: flex;
  justify-content: center;
  background-color: ${whiteColor};
  height: 64px;
  width: 100%;
`;

export const Content = styled.div`
  width: 100%;
  max-width: ${containerSize};
  display: flex;
  justify-content: space-between;
`;

export const LogoContainer = styled.div`
  width: fit-content;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const MainLogo = styled.img`
  width: 133px;
  height: 26px;
  object-fit: contain;
`;

export const RoutesContainer = styled.div`
  width: 50%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const RouteContainer = styled.div`
  height: 100%;
  width: fit-content;
  display: flex;
  flex-direction: column;
  justify-content: center;

  &.active {
    border-top: 4px solid ${primaryColor};
  }

  &.active > a {
    margin-top: -8px;
  }
`;

export const Route = styled.a`
  font-family: 'Roboto', sans-serif;
  font-size: 14px;
  font-weight: bold;
  letter-spacing: 0.3px;
  color: ${primaryColor};
  cursor: pointer;
`;
