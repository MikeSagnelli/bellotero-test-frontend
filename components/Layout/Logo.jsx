import React from 'react';
import { LogoContainer, MainLogo } from './styles';

const Logo = () => (
  <LogoContainer>
    <MainLogo src="/static/images/bellotero.svg" alt="Bellotero" />
  </LogoContainer>
);

export default Logo;
