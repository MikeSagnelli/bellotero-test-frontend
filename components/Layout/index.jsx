import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { connect } from 'react-redux';
import { MainContainer, Header, Content, RoutesContainer, RouteContainer, Route } from './styles';
import Logo from './Logo';

const Layout = ({ children, globalData, activeRoute }) => (
  <MainContainer>
    <Header>
      <Content>
        <Logo />
        <RoutesContainer>
          {globalData.menu.items.map(route => (
            <RouteContainer
              className={activeRoute === route.route ? 'active' : ''}
              key={route.text}
            >
              {route.route !== '#' ? (
                <Link as={route.text} href={route.route}>
                  <Route id="no-route">{route.text}</Route>
                </Link>
              ) : (
                <Link href={route.route}>
                  <Route id="route">{route.text}</Route>
                </Link>
              )}
            </RouteContainer>
          ))}
        </RoutesContainer>
      </Content>
    </Header>
    {children}
  </MainContainer>
);

Layout.displayName = 'Layout';

Layout.propTypes = {
  children: PropTypes.node,
  globalData: PropTypes.shape({
    menu: PropTypes.shape({
      items: PropTypes.arrayOf(
        PropTypes.shape({
          text: PropTypes.string,
          route: PropTypes.string
        })
      )
    })
  }),
  activeRoute: PropTypes.string
};

Layout.defaultProps = {
  children: null,
  globalData: null,
  activeRoute: ''
};

export default connect(state => state)(Layout);
