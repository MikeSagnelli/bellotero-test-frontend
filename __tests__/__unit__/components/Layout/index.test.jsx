import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../../../../components/Layout';

describe('Test Layout', () => {
  it('renders layout', () => {
    const component = shallow(<Layout />);
    expect(component).toMatchSnapshot();
  });
});
