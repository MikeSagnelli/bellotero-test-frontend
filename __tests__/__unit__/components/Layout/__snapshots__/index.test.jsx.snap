// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout renders layout 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
